### PR TITLE
Feat/compliance tool drop versions add tool array

### PR DESCRIPTION
**Pull request by garloff**

### CODE REVIEW

**Comment by garloff** [comment]
Branch is relative to feat/compliance-tool-drop-versions branch which is relative to feat/compliance-tool branch.
Note: There is an open question whether condition should be per check_tool and not per standard.

File: Design-Docs/tools/scs-compatible.yaml [hunk 1]
```diff
@@ -6,11 +6,13 @@ iaas:
     standards:
       - name: Flavor naming
         url: https://raw.githubusercontent.com/SovereignCloudStack/Docs/main/Standards/SCS-0003-v2-flavor-naming.md
-        check_tool: flavor-names-openstack.py
+        check_tools:
+          - url: flavor-names-openstack.py
```
Comment (tibeer) [inline Design-Docs/tools/scs-compatible.yaml:10]: _url_ is somewhat misleading if you are using local files like here. Maybe rather _uri_ or _source_?

File: Standards/scs-xxxx-v1-sovereign-cloud-standards-yaml.md [hunk 2]
```diff
@@ -127,11 +128,15 @@ iaas:
     standards:
       - name: Flavor naming
         url: https://raw.githubusercontent.com/SovereignCloudStack/Docs/main/Standards/SCS-0003-v1-flavor-naming.md
-        check_tool: flavor-name-check.py
+        check_tools: 
+	  - url: flavor-name-check.py
```
Comment (tibeer) [inline Standards/scs-xxxx-v1-sovereign-cloud-standards-yaml.md:132]: This seems to be a tab-stop

**Review by tibeer — CHANGES_REQUESTED** [review]

File: Design-Docs/tools/scs-compatible.yaml (hunk 1 above)
Comment (garloff) [reply Design-Docs/tools/scs-compatible.yaml:10]: Good point. `executable` would seem to best describe what this is. Something that can be run...
Opinions?

File: Standards/scs-xxxx-v1-sovereign-cloud-standards-yaml.md (hunk 2 above)
Comment (garloff) [reply Standards/scs-xxxx-v1-sovereign-cloud-standards-yaml.md:132]: Looks like vi misconfiguration. Thanks for catching this one!

**Comment by garloff** [comment]
> Note: There is an open question whether condition should be per check_tool and not per standard.

We could even allow both.
Per standard would set the default for all check_tools belonging to the standard and it could be overridden per tool. 

Opinions?

File: Design-Docs/tools/scs-compatible.yaml (hunk 1 above)
Comment (tibeer) [reply Design-Docs/tools/scs-compatible.yaml:10]: This!

**Comment by tibeer** [comment]
> > Note: There is an open question whether condition should be per check_tool and not per standard.
> 
> We could even allow both. Per standard would set the default for all check_tools belonging to the standard and it could be overridden per tool.
> 
> Opinions?

No strong opinion. Would leave it to you and your current workload/motivation on implementing it.

**Comment by garloff** [comment]
Ready for re-review, @tibeer .
- I rename `url` to `executable`.
- I allow for `condition` per `check_tool` to override the `condition` from the `standard` (even if I believe we'll not need it often).
Hopefully, this is now very agreeable.

**Review by tibeer — APPROVED** [review]
LGTM!

**Review by itrich — APPROVED** [review]
Works fine for me! 👍

```
itrich@Tuxi:~/Git/SCS/Docs/Design-Docs/tools$ OS_CLOUD=gxscs-capi ./scs-compliance-check.py scs-compatible.yaml iaas -o report.yaml
Testing iaas standard version v1
*******************************************************
Testing standard Flavor naming ...
Reference: https://raw.githubusercontent.com/SovereignCloudStack/Docs/main/Standards/SCS-0003-v1-flavor-naming.md ...

... returned 0 errors

... returned 0 errors
*******************************************************
Testing standard Image metadata ...
Reference: https://raw.githubusercontent.com/SovereignCloudStack/Docs/main/Standards/SCS-0004-v1-image-metadata.md ...
Warning: Image "Ubuntu Minimal 22.04" does not start with recommended name "ubuntu 22.04"
Warning: Image "Ubuntu Minimal 20.04" does not start with recommended name "ubuntu 20.04"
INFO: Recommended image "CentOS 8" is missing
INFO: Recommended image "Rocky 8" is missing
INFO: Recommended image "Fedora 36" is missing
INFO: Recommended image "SLES 15SP4" is missing
INFO: Recommended image "RHEL 9" is missing
INFO: Recommended image "RHEL 8" is missing
INFO: Recommended image "Windows Server 2022" is missing
INFO: Recommended image "Windows Server 2019" is missing

... returned 0 errors
*******************************************************
Testing standard OpenStack Powered Compute v2020.11 ...
Reference: https://opendev.org/openinfra/interop/src/branch/master/guidelines/2020.11.json ...
WARNING: No compliance check tool implemented yet for OpenStack Powered Compute v2020.11
*******************************************************
Verdict for os_cloud gxscs-capi, layer iaas, version v1: PASSED
```
with output

```
name: SCS Compatible
url: https://raw.githubusercontent.com/SovereignCloudStack/Docs/main/Design-Docs/tools/scs-compatible.yaml
os_cloud: gxscs-capi
checked_at: 2023-01-31
iaas:
- version: v1
  stabilized_at: 2021-01-01
  obsoleted_at: 2023-10-03
  standards:
  - name: Flavor naming
    url: https://raw.githubusercontent.com/SovereignCloudStack/Docs/main/Standards/SCS-0003-v1-flavor-naming.md
    condition: mandatory
    check_tools:
    - executable: flavor-names-openstack.py
      errors: 0
    - executable: flavor-name-check.py
      args: SCS-1V:4:10
      condition: optional
      errors: 0
  - name: Image metadata
    url: https://raw.githubusercontent.com/SovereignCloudStack/Docs/main/Standards/SCS-0004-v1-image-metadata.md
    check_tools:
    - executable: image-md-check.py
      args: -v
      errors: 0
    condition: mandatory
  - name: OpenStack Powered Compute v2020.11
    url: https://opendev.org/openinfra/interop/src/branch/master/guidelines/2020.11.json
    condition: mandatory
  errors: 0
```

**Pull request merged**